rook-ceph/README.md

```diff
@@ -0,0 +1,452 @@
+# Use Rook Ceph on Azure Kubernetes Service
+
+[Rook](https://github.com/rook/rook) is an open source cloud-native storage orchestrator for Kubernetes.
+It turns existing storage software into self-managing, self-scaling, and self-healing storage services that run seamlessly on-top of Kubernetes.
+It is a [CNCF graduated](https://www.cncf.io/projects/) project.
+
+[Ceph](https://ceph.com/en/) is a highly scalable distributed storage solution for block storage, object storage, and shared filesystems with years of production deployments.
+
+This article shows you how to deploy Rook Ceph on Azure Kubernetes Service (AKS) and use it as the storage solution for your cluster.
+
+## Before you begin
+
+This article assumes that you have an existing Azure subscription and installed Azure CLI. If you haven't done so, see [how to install Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli).
+
+## Step 1: Create cluster and node pools
+
+### Step 1.1: Create a new AKS cluster with a system node pool
+
+To run production workloads, create a system node pool with at least 3 nodes.
+
+```bash
+# Create a resource group.
+export RESOURCE_GROUP="${USER}ResourceGroup"
+az group create --name ${RESOURCE_GROUP} --location eastus
+
+# Create a cluster.
+export CLUSTER="${USER}Cluster"
+az aks create -g ${RESOURCE_GROUP} --name ${CLUSTER} --node-count 3 --generate-ssh-keys
+
+# Get credentials for cluster access.
+az aks get-credentials -g ${RESOURCE_GROUP} --name ${CLUSTER} --overwrite-existing
+```
+
+References:
+* [Quickstart: Deploy an Azure Kubernetes Service cluster using the Azure CLI](https://docs.microsoft.com/en-us/azure/aks/kubernetes-walkthrough)
+
+### Step 1.2: Create a storage node pool
+
+To isolate storage from your applications, create a dedicated node pool for Rook Ceph.
+
+```bash
+az aks nodepool add -g ${RESOURCE_GROUP} --cluster-name ${CLUSTER} --name storagepool --node-count 3 --node-taints storage-node=true:NoSchedule --labels storage-node=true
+```
+
+References:
+* [Create and manage multiple node pools for a cluster in Azure Kubernetes Service (AKS)](https://docs.microsoft.com/en-us/azure/aks/use-multiple-node-pools)
+
+### Step 1.3: Verify
+
+Verify that the nodes are in the `Ready` state before proceeding.
+
+```bash
+kubectl get nodes
+```
+
+Expected output:
+```
+NAME                                   STATUS   ROLES   AGE   VERSION
+aks-nodepool1-14514606-vmss000000      Ready    agent   20m   v1.20.7
+aks-nodepool1-14514606-vmss000001      Ready    agent   20m   v1.20.7
+aks-nodepool1-14514606-vmss000002      Ready    agent   20m   v1.20.7
+aks-storagepool-14514606-vmss000000    Ready    agent   94s   v1.20.7
+aks-storagepool-14514606-vmss000001    Ready    agent   74s   v1.20.7
+aks-storagepool-14514606-vmss000002    Ready    agent   93s   v1.20.7
+```
+
+## Step 2: Deploy Rook Ceph
+
+### Step 2.1: Clone this repo
+
+```bash
+git clone https://github.com/Azure/kubernetes-volume-drivers
```

Comment:
`git clone` is not necessary if the example commands are all using `https://raw.githubusercontent.com/...` path, that would be easier in deployment

Reply:
I intentionally used https://raw... for files from rook and used git clone for files from this repo, as I expect customers to modify files from this repo, e.g., changing the storage capacity for OSDs.